guidance_plugin_validator/src/guidance_plugin_validator/guidance_plugin_components.py

```diff
@@ -0,0 +1,402 @@
+            rospy.loginfo("Success: " + str(self.plugin_name) + " Has correct plugin discovery 'capability' " + str(self.requirement_results.correct_plugin_discovery_capability))
+        else:
+            rospy.logerr("ERROR: " + str(self.plugin_name) + " Has incorrect plugin discovery 'capability' " + str(self.requirement_results.correct_plugin_discovery_capability))
+
+        # Output results for optional plugin components
+        rospy.loginfo("Optional Components: ")
+
+        if self.optional_results.has_current_pose_sub:
+            rospy.loginfo("Success: " + str(self.plugin_name) + " Subscribes to " + str(self.optional_results.current_pose_topic))
+        else:
+            rospy.logwarn("WARNING: " + str(self.plugin_name) + " Does not subscribe to " + str(self.optional_results.current_pose_topic))
+        
+        if self.optional_results.has_current_speed_sub:
+            rospy.loginfo("Success: " + str(self.plugin_name) + " Subscribes to " + str(self.optional_results.current_speed_topic))
+        else:
+            rospy.logwarn("WARNING: " + str(self.plugin_name) + " Does not subscribe to " + str(self.optional_results.current_speed_topic))
+        
+        if self.optional_results.has_plugin_discovery_available:
+            rospy.loginfo("Success: " + str(self.plugin_name) + " Has plugin discovery message 'available' set to True")
+        else:
+            rospy.logwarn("WARNING: " + str(self.plugin_name) + " Does not have plugin discovery message 'available' field set to True.")
+
+        if self.optional_results.has_plugin_discovery_activated:
+            rospy.loginfo("Success: " + str(self.plugin_name) + " Has plugin discovery message 'activated' field set to True")
+        else:
+            rospy.logwarn("WARNING: " + str(self.plugin_name) + " Does not have plugin discovery message 'activated' field set to True.")
+
+class RequiredTacticalComponents():
+    """
+    Convenience class used to store the validation results for components of a tactical plugin
+    designated as required.
+    """
+
+    def __init__(self, plugin_name):
+        """Constructor for RequiredTacticalComponents"""
+
+        # Validation result indicating whether tactical plugin's node successfully launches
+        self.has_node = False
+
+        # Validation result indicating whether tactical plugin's node advertises required service
+        self.plan_trajectory_service = "/guidance/plugins/" + str(plugin_name) + "/plan_trajectory"
+        self.has_plan_trajectory_service = False
+
+        # Validation result indicating whether tactical plugin's node publishes to required topic
+        self.plugin_discovery_topic = "/guidance/plugin_discovery"
+        self.has_plugin_discovery_pub = False
+
+        # Validation results indicating whether tactical plugin's node publishes required information to the plugin_discovery topic
+        self.correct_plugin_discovery_type = 2
```

Comment:
Not sure what this integer is representing exactly, same as the above usage for strategic. If possible can we replace this with an IntEnum that provides some more clarity?

Reply:
Fixed.